app/models/allocation.rb

```diff
@@ -92,6 +92,14 @@ def refresh_status_and_moves_count!
     save!
   end
 
+  def move_totals
```

Comment:
is there a way we can somehow adapt `self.move_totals` to support both a single and multiple allocations to avoid duplication?

Reply:
Or maybe the instance method can invoke the class method on an active record relation from the single allocation

Reply:
Its's a good question and is something I considered but decided against as although the logic for the the metadata rules is the same in both single and multiple cases, the implementation has to be widely different to be performant. The class version of this method returns a hash of allocation id's for each set of metadata, so would need quite a bit of munging both in calling and consuming the class method for it to be shared, and overall that would look worse and potentially more confusing than keeping them separate.